Functions/BIDS/xASL_bids_MergeNifti_Merge.m

```diff
@@ -75,17 +83,30 @@
                 end
             end
         end
+        
+	end
 
-        % Check for the path to JSON if existing and keep only the first existing JSON
-        if isempty(firstJSON)
-            [Fpath, Ffile] = xASL_fileparts(NiftiPaths{indexSortedFile(iFile)});
-            pathJSON = fullfile(Fpath,[Ffile '.json']);
-            if exist(pathJSON, 'file')
-                firstJSON = pathJSON;
-            end
-        end
-    end
-
+	%% Merge all the JSONs according to their priority
+	
+	% Create an order in which the JSONs are going to be read
+	[~, priorityListIndex] = sort(priorityList);
+	outputJSON = struct();
+	for iJSON = 1:length(NiftiPaths)
+		% Go through the JSONs in increasing priority
+		[Fpath, Ffile] = xASL_fileparts(NiftiPaths{priorityListIndex(iJSON)});
+		if xASL_exist(fullfile(Fpath,[Ffile '.json']), 'file')
+			currentJSON = spm_jsonread(fullfile(Fpath,[Ffile '.json']));
+			
+			% Go through all fields
+			listFieldNames = fieldnames(currentJSON);
+			for iFieldName = 1:length(listFieldNames)
```

Comment:
Is this field-wise compiling of the merged JSON the best solution? Why doing this instead of just prioritizing one full JSON that dcm2nii outputted?

(not saying it's not a good idea, just saying that this may or may not work. One disadvantage I can think of, is that you potentially get a mix of values belonging to a label and an M0 image).

Reply:
I'm not really sure either - previously, I was always taking a random JSON, because there were no differences between them. This is the first case encountered that they actually differ :) 

So hard to say if it is better to select one or compile all :) Or even impossible to say - both can be later proven to be wrong :) For now, I would slightly prefer merging (and checking if there's nothing extra), then selecting one and potentially missing something. 

But if you think that selecting is better, I am not against ;)

Reply:
I would at least issue warnings if there are differences between the JSONs. Then I would choose the priority JSON. Then you can resolve :)

Reply:
I've added that for LabDur and PLD - and we can add more important fields. But otherwise this makes no sense. There are a lot of differences between JSONs in parameters not important for ASL. It would be issuing to many warnings if warning for all.